src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/DnsZone/New-ArmDnsZoneARecord.ps1

```diff
@@ -38,6 +38,6 @@ function New-ArmDnsZoneARecord {
         }
 
         $aRecord.PSTypeNames.Add("ArmResource")
-        return $aRecord
+        return $aRecord | Add-ArmDependencyOn -Dependency $PrivateDnsZone -PassThru
```

Comment:
I suggest we change this cmdlet to `Add-ArmDnsZoneARecord` and return the DnsZone here instead of the ARecord